packages/pocket-ic/src/common/rest.rs

```diff
@@ -861,13 +861,15 @@ pub struct RawMockCanisterHttpResponse {
     pub subnet_id: RawSubnetId,
     pub request_id: u64,
     pub response: CanisterHttpResponse,
+    pub additional_responses: Option<Vec<CanisterHttpResponse>>,
 }
 
 #[derive(Clone, Serialize, Deserialize, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
 pub struct MockCanisterHttpResponse {
     pub subnet_id: Principal,
     pub request_id: u64,
     pub response: CanisterHttpResponse,
+    pub additional_responses: Option<Vec<CanisterHttpResponse>>,
```

Comment:
Exactly! I know it may feel a bit weird, but I wanted to easily support the case of a single response (replicated behind the scenes).

Reply:
We could also have a single field `responses: Vec<CanisterHttpResponse>`, but then it's less clear what to do if you only want to supply a single response (we could check if the length of the vector is one and replicate the single response, but it looks more weird in my opinion and this is likely the main use case to have a single response).

Reply:
It seems like `Some(vec![])` would be rejected, so why not just require `Vec<..>`? I think writing `None` versus `vec![]` is not that much more convenient. 

It's a bit strange anyway to split the two args. Are you aiming for backwards compatibility? Because I think we don't have that here anyway. 

How about just `responses: Vec<CanisterHttpResponse>` and we just document clearly that it must be non-empty and if it contains more than one entry, then up to <subnet_size> of them will be "used", dropping the constraint of "either one or <subnet_size>". TBH, I don't find this suggestion much better, just sharing my thoughts.

Reply:
Our comments collided in time

Reply:
I'll leave the decision to you. But if you keep it, the comment in the changelog could be made a bit clearer IMO.

Reply:
> Are you aiming for backwards compatibility? Because I think we don't have that here anyway.

We have that here since you don't need to provide optional fields.

Reply:
> we could distribute the N < subnet_size given responses to the M = subnet_size slots for the user

Indeed, but it is not clear which of the responses (that could be pairwise distinct) to replicate.

> not for the Rust library users

you're right, this is because we expose the entire struct to the Rust library user, but now it's too late for the sake of this PR.